app/workers/notification_workers/emotion_selection_notification_worker.rb

```diff
@@ -0,0 +1,17 @@
+module NotificationWorkers
+  class EmotionSelectionNotificationWorker
+
+    attr_reader :users, :time_period
+
+    def initialize
+      @users = User.opt_in
+      @time_period = TimePeriod.last
```

Comment:
Idea: what about creating a model method in `TimePeriod`, something like `TimePeriod.current` and then in that method compare the `start_date` and `end_date`? e.g.
```ruby
def self.current
  TimePeriod.find_by(start_date: ..Date.current, end_date: Date.current..)
end
```
If for some reason there isn't a new `time_period` record created for the current week, we don't want emails to go out for a previous week.